### ISSUE TITLE
Use Docker API to import image instead of shared volumes in executable.

### ISSUE THREAD

**Issue by j-wilkins** [asaaki/elixir-docker-image-packager#3]
first off, thanks for this! It looks great, and when I run it inside the stage container manually it works awesome. Very nice work.

Do you have any interest in using the docker API instead of adding the `docker` executable and `docker.sock` as volumes?

I could be wrong, but I think that when running this under `Boot2Docker` or something similar mounting those fails because those docker solutions can only share stuff that's located in `$HOME`. At least, running it under those conditions fails for me with a `read only filesystem` type error.

I was able to get the image imported without mounting the `docker` stuff as volumes by `curl`ing the tarball like:

```
 curl --data-binary "@/stage/tarballs/20150803_041551/silo-0.0.1.tar.gz" -H 'Content-Type: application/tar' -X POST "http://192.168.0.34:2376/images/create?fromSrc=-&changes=CMD%20trap%20exit%20TERM%3B%20/app/bin/silo%20foreground&repo=local/silo:0.0.1"
```

where the host bit can be read from `$DOCKER_HOST`

Any interest? Obviously it has its trade offs as well, so no worries if this isn't a direction you want to head.

💕

**Comment by asaaki** [comment]
I'm aware of the Boot2Docker issue, because I'm running most of the development stuff on a Mac, too.

Therefore I decided to build the stuff in the scope of my home directory.

What you can do is to symlink your non-HOME directory into your HOME and build from there.

``` shell
ln -s <location of your project> $HOME/<your project>
```

Or what I generally do: create a symlink of your development directory into HOME:

``` shell
ln -s /my/development/directory $HOME/Development
```

I don't really want to "fix" this by using the Docker API, because this is a very Mac-specific issue and Linux users do not have the problem. With the workaround described here also Mac users should still be able to create their images. (Ideally Boot2Docker will fix their issue with custom vbox shares, which should then allow any location besides the home folder.)

**tl;dr** Not right now. My current focus is on stabilizing, fixing corner-cases, allowing better builds for Phoenix apps. Platform-specific issues might follow later when I think the project itself is in a good shape.

**Comment by asaaki** [comment]
I tried the symlinking. Unfortunately the tarball saving does not work then, because the symlinked path is resolved and points to the non-HOME directory.

So only solution is then to move the project to a location within your HOME path and build from there.

**Comment by j-wilkins** [comment]
Hey, thanks for looking at this!

My project is in my $HOME, but when the top level Makefile boots the stage image it uses `env docker` and mounts that as a volume, which returns something in `/usr/local/bin`, same with the `docker.sock`, and that's what is causing my issue.

I'm totally with you though that this is a platform specific thing and can wait, I can still get it to run by booting the stage container without those volumes and generating the tarball. 

Best,
Jake

**Comment by asaaki** [comment]
Do you use `boot2docker` directly?
Or [`docker-machine`](https://github.com/docker/machine) by any chance?
Just asking, because with docker-machine the volumes for docker (bin and socket) work just fine on my machine. (docker-machine uses boot2docker on OSX under the hood.)

**Comment by j-wilkins** [comment]
Well that’ll teach me to think I know what’s going on.

It works fine in `boot2docker`, I was trying to run it using the docker instance on `coreos-vagrant` and I thought I knew enough about how they were both using virtual box under the hood. But apparently I don’t. :)

Thanks for taking the time to troubleshoot, sorry that I could have saved us both some trouble if I wasn’t trying to do things the hard way.

**Comment by asaaki** [comment]
No problem at all, you're welcome.

The Docker API is still something I should have a look into. So, thanks for the reminder! :+1:
